tools/harness-automation/autothreadharness/runner.py

```diff
@@ -106,10 +106,17 @@ def add_result(self, test, passed, error=None):
             test (TestCase): The test just run
             passed (bool): Whether the case is passed
         """
+        if str(test.__class__.__name__) in self.result and 'fails' in self.result[str(test.__class__.__name__)]:
```

Comment:
is `str(test.__class__.__name__)` necessary? I believe `test.__class__.__name__` is assured to be `str`